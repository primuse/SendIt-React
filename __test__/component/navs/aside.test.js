import React from 'react';
import renderer from 'react-test-renderer';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import ConnectedAside from '../../../src/components/navs/aside.jsx';

const initialState = {
  auth: {
    isAuthenticated: true,
    user: {
      firstName: 'Tiku',
      lastName: 'Okoye',
      isadmin: false
    },
  }
};
const mockStore = configureStore([thunk]);
const store = mockStore(initialState);
let wrapper;


describe('Aside nav component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<MemoryRouter>
        <Provider store={store}>
          <ConnectedAside />
        </Provider>
      </MemoryRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Aside Nav component', () => {
  beforeEach(() => {
    wrapper = mount(<MemoryRouter>
      <Provider store={store}><ConnectedAside /></Provider>
    </MemoryRouter>);
    wrapper.instance().showModal = jest.fn();
    wrapper.instance().closeModal = jest.fn();
  });
  afterEach(() => {
    wrapper.unmount();
  });
  it('should simulate the opening of the new parcel modal', () => {
    wrapper.update();
    const aTag = wrapper.find('a').at(2);
    aTag.simulate('click');
    // expect(wrapper.instance().showModal).toHaveBeenCalled();
  });
});
