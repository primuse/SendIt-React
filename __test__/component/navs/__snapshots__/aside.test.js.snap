// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Aside nav component renders correctly 1`] = `
<aside
  className=""
>
  <div
    className="bg-blue"
    id="menu"
  >
    <div
      id="aside-profile"
    >
      <a
        href="/dashboard"
        onClick={[Function]}
      >
        <h3
          className="m-0 pb-10"
        >
          Send
          <span
            className="head-span"
          >
            IT
          </span>
        </h3>
      </a>
      <small
        className="d-block pb-10 white-3"
        id="user-id"
      >
        SD0
      </small>
      <small>
        <p
          className="m-0"
        >
          Welcome 
        </p>
        <p
          id="username"
        >
           
        </p>
      </small>
    </div>
    <ul
      className="mt-80"
    >
      <li>
        <img
          className="menu-ico"
          src="test-file-stub"
        />
        <a
          href="/dashboard"
          onClick={[Function]}
        >
          <span>
            Dashboard
          </span>
        </a>
      </li>
      <li>
        <img
          className="menu-ico"
          src="test-file-stub"
        />
        <a
          data-modal={true}
          data-target="#parcelmodal"
          href="/"
          onClick={[Function]}
        >
          <span>
            New Parcel
          </span>
        </a>
      </li>
      <li>
        <img
          className="menu-ico"
          src="test-file-stub"
        />
        <a
          href="/delivered_parcels"
          onClick={[Function]}
        >
          <span>
            Delivered Parcels
          </span>
        </a>
      </li>
      <li>
        <img
          className="menu-ico"
          src="test-file-stub"
        />
        <a
          href="/inTransit_parcels"
          onClick={[Function]}
        >
          <span>
            In-transit Parcels
          </span>
        </a>
      </li>
    </ul>
  </div>
</aside>
`;
